src/typescript/providers/src/wix/wix.provider.ts

```diff
@@ -0,0 +1,171 @@
+
+        const wixAccountId = input.getVariable(this.config.accountId);
+        if (wixAccountId == null) {
+            throw new Error('Wix provider: accountId not provided');
+        }
+
+        const wixDnsZone = input.getVariable(this.config.dnsZone);
+        if (wixDnsZone == null) {
+            throw new Error('Wix provider: dnsZone not provided');
+        }
+
+        const recordType = input.ipVersion === 'IPv4' ? 'A' : 'AAAA';
+
+        const hosts = Array.from(new Set([
+            ...this.config.hosts ?? [],
+            ...this.config.useHostsFromRequest === true ? input.requestHosts : [],
+        ]));
+
+        for (const wixHost of hosts) {
+            const host = typeof wixHost === 'string' ? wixHost : wixHost.name;
+            if (host !== wixDnsZone && !host.endsWith(`.${wixDnsZone}`)) {
+                throw new CustomError({
+                    message: `Wix provider: host ${host} should end in ${wixDnsZone}`,
+                });
+            }
+        }
+
+        const getDnsZoneResponse = await this.wixRequest(
+            `domains/v1/dns-zones/${wixDnsZone}`,
+            wixApiKey,
+            wixAccountId,
+        );
+        if (!getDnsZoneResponse.ok) {
+            throw new CustomError({
+                message: 'Wix provider: Failed to list dns records',
+                info: {
+                    zone: wixDnsZone,
+                    responseData: await getDnsZoneResponse.json(),
+                },
+            });
+        }
+
+        const getDnsZoneResponseData: {dnsZone: WixDnsZone} = await getDnsZoneResponse.json();
+        const dnsRecords = getDnsZoneResponseData.dnsZone.records
+            .filter(record => record.type === recordType);
+
+        const additions = new Array<WixDnsRecord>();
+        const deletions = new Array<WixDnsRecord>();
+
+        for (const wixHost of hosts) {
+            const isWixAdvancedHost = typeof wixHost !== 'string';
+            const host = isWixAdvancedHost ? wixHost.name : wixHost;
+            const existingDnsRecord = dnsRecords.find(record => record.hostName === host);
+
+            let newTtl: number | undefined;
+            if (isWixAdvancedHost && wixHost.ttl != undefined) {
+                newTtl = wixHost.ttl;
+            } else if (existingDnsRecord !== undefined) {
+                newTtl = existingDnsRecord.ttl;
+            }
+
+            additions.push({
+                hostName: host,
+                ttl: newTtl,
+                type: recordType,
+                values: [input.ip],
+            });
+            if (existingDnsRecord !== undefined) {
+                deletions.push({
+                    hostName: host,
+                    ttl: existingDnsRecord.ttl,
+                    type: recordType,
+                    values: existingDnsRecord.values,
+                });
+            }
+        }
+
+        const updateDnsResponse = await this.wixRequest(
+            `domains/v1/dns-zones/${wixDnsZone}`,
```

Comment:
Can you add a link to <https://dev.wix.com/docs/rest/account-level/domains/domain-dns/update-dns-zone> in a comment?